app/experimenter/experiments/tests/test_forms.py

```diff
@@ -601,6 +604,32 @@ def test_form_removes_variant(self):
             ).exists()
         )
 
+    def test_form_checks_uniqueness_for_single_experiment_not_all(self):
+        # We want to make sure that all the branches in a single
+        # experiment have unique names/slugs but not that they're
+        # unique across all experiments.  This behaviour was accidentally
+        # introduced in 30c210183cd1568850d54132633b0f23e3e56c98
+        # so I'm adding a test for it.
+        experiment1 = ExperimentFactory.create_with_status(
+            Experiment.STATUS_DRAFT, num_variants=0
+        )
+        form1 = ExperimentVariantsAddonForm(
+            request=self.request, data=self.data, instance=experiment1
+        )
+        self.assertTrue(form1.is_valid())
+        experiment1 = form1.save()
+        self.assertEqual(experiment1.variants.count(), 3)
+
+        experiment2 = ExperimentFactory.create_with_status(
+            Experiment.STATUS_DRAFT, num_variants=0
+        )
+        form2 = ExperimentVariantsAddonForm(
+            request=self.request, data=self.data, instance=experiment2
```

Comment:
This works because `self.data` is used for both experiments right? Is that where the branch names are?

Reply:
Yeah exactly, the dict is read, not modified, and the goal is to create two experiments with identical branch naming (which was causing the collision).